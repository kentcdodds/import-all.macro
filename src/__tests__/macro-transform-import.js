import path from 'path'
import plugin from 'babel-plugin-macros'

import {createMacroTests} from './helpers/create-macro-test'

createMacroTests({
  plugin: (babel, options) => {
    return plugin(babel, {
      importAll: {
        transformModulePath(modulePath, importingPath) {
          const projectRoot = path.join(__dirname, '../../')
          const modulePathWithoutExt = modulePath.replace(/\.js$/, '')
          const absolutePath = path.resolve(
            path.dirname(importingPath),
            modulePathWithoutExt,
          )
          const pathRelativeToRoot = path.relative(projectRoot, absolutePath)
          return pathRelativeToRoot
        },
      },
      ...options,
    })
  },
  babelOptions: {
    filename: __filename,
  },
  tests: {
    'README:4 configure `importAll` to transform import path before generating imports': `
      import importAll from '../macro'

      const a = importAll.sync('./fixtures/*.js')
    `,
  },
})
