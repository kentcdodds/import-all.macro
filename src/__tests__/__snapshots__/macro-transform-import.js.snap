// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros README:4 configure \`importAll\` to transform import path before generating imports: README:4 configure \`importAll\` to transform import path before generating imports 1`] = `

import importAll from 'import-all.macro'

const a = importAll.sync('./files/*.js')

      ↓ ↓ ↓ ↓ ↓ ↓

import * as _src__tests__FixturesA from 'src/__tests__/files/a'
import * as _src__tests__FixturesB from 'src/__tests__/files/b'
import * as _src__tests__FixturesC from 'src/__tests__/files/c'
import * as _src__tests__FixturesD from 'src/__tests__/files/d'
const a = {
  'src/__tests__/files/a': _src__tests__FixturesA,
  'src/__tests__/files/b': _src__tests__FixturesB,
  'src/__tests__/files/c': _src__tests__FixturesC,
  'src/__tests__/files/d': _src__tests__FixturesD,
}


`;
